src/hooks/useWithoutAnonLoginOptions.ts

```diff
@@ -0,0 +1,37 @@
+import { getChatIdsWithoutAnonLoginOptions } from '@/constants/chat'
+import { useLoginModal } from '@/stores/login-modal'
+import { useMyAccount } from '@/stores/my-account'
+import { generateManuallyTriggeredPromise } from '@/utils/promise'
+import { getIdFromSlug } from '@/utils/slug'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useRef } from 'react'
+
+export default function useWithoutAnonLoginOptions() {
```

Comment:
Rename to useLoginOptions 

and return 

isLoginRequired
promptUserForLogin

Reply:
For future it will be main option, so better call it more generic I guess

Reply:
updated to isNonAnonLoginRequired
because isLoginRequired is pretty ambiguous to me